### PR TITLE
faucet-pipeline-static: Document the new `filter` option

### PR DESCRIPTION
**Pull request by mkhl**

### CODE REVIEW

File: static.md [hunk 1]
```diff
@@ -20,3 +20,22 @@ module.exports = {
     }]
 };
 ```
+
+If you only want to copy _some_ of the files,
+you can select them using a `filter` predicate.
```
Comment (FND) [inline static.md]: I'm not sure our target audience is comfortable with the use of "predicate" here. @moonglum, do we have an established term for this?

File: static.md [hunk 2]
```diff
@@ -20,3 +20,22 @@ module.exports = {
...
+The predicate will be called for every file with its path relative to `source`.
+Each file it returns `true` for gets copied.
```
Comment (FND) [inline static.md]: This seems a little hard to parse; perhaps this scans better:

> Only files for which that function returns `true` are copied.

Alternatively:

> Any return value other than `true` will result in the respective file being omitted / _not_ being copied.

File: static.md [hunk 3]
```diff
@@ -20,3 +20,22 @@ module.exports = {
...
+        source: "./images",
+        target: "./public/images",
+        filter: file => !file.endsWith(".psd")
+    }, {
+        source: "./fonts",
+        target: "./public/fonts",
+        filter: file => file.endsWith(".ttf")
+    }]
+}
+```
```
Comment (FND) [inline static.md]: A sentence like this would reinforce understanding:

> In the example above, `.psd` images will be omitted and only `.ttf` fonts will be copied.

**Comment by moonglum** [comment]
@mkhl Woha... I've not seen this since march?! What other things have I missed oO

File: static.md (hunk 1 above)
Comment (moonglum) [reply static.md]: I would go with the term 'predicate'. We can also go with 'function' if you think that 'predicate' is not well established among our users. But I would just go with 'predicate' and wait if anyone has a problem with it.

Comment (mkhl) [reply static.md]: I believe this may be what they call “prioritising” or “wanting to ship the damn thing”.

How about an appeal to authority?
[The MDN docs on `Array.prototype.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) avoid the term “predicate”.  Instead they say:
> The filter() method creates a new array with all elements that pass the test implemented by the provided function.

File: static.md [hunk 4]
```diff
@@ -20,3 +20,28 @@ module.exports = {
...
+In this example,
+we omit images in the `templates/` directory
+and only copy `.ttf` fonts.
+
+```js
+module.exports = {
+    static: [{
+        source: "./images",
+        target: "./public/images",
+        filter: file => !file.startsWith("templates/")
```
Comment (FND) [inline static.md]: As a naive user, I just realized that starting with a negated unnecessarily introduces some latent uncertainty. (Not a lot, of course, but Cheating God etc.)

Comment (mkhl) [reply static.md]: I’m not sure whether this comment is informative or appellative in nature.

Comment (FND) [reply static.md]: How about passive-aggressive?

Comment (FND) [reply static.md]: What do you think of 9f9516d80e5db89bddf8197013d7cf6fa003b42c? Feel free to reject it.

Comment (mkhl) [reply static.md]: Looks good to me

**Comment by FND** [comment]
Squashed as 6d71087a6aae89033859b487343bac7e0efdec82 - thank you!
